src/interfaces/swig/Library.i

```diff
@@ -26,312 +23,53 @@
-    %template(IntNDArray)  SGNDArray<int32_t>;
-#endif
-#ifdef USE_UINT32
-    %template(UIntNDArray)  SGNDArray<uint32_t>;
-#endif
-#ifdef USE_INT64
-    %template(LongIntNDArray)  SGNDArray<int64_t>;
-#endif
-#ifdef USE_UINT64
-    %template(ULongIntNDArray)  SGNDArray<uint64_t>;
-#endif
-#ifdef USE_FLOAT32
-    %template(ShortRealNDArray) SGNDArray<float32_t>;
-#endif
-#ifdef USE_FLOAT64
-    %template(RealNDArray) SGNDArray<float64_t>;
-#endif
-#ifdef USE_FLOATMAX
-    %template(LongRealNDArray) SGNDArray<floatmax_t>;
-#endif
-}
-
-/* Include Class Headers to make them visible from within the target language */
-/* Template Class DynamicArray */
-%include <shogun/lib/DynamicArray.h>
-namespace shogun
-{
-#ifdef USE_CHAR
-        %template(DynamicCharArray) DynamicArray<char>;
-#endif
-#ifdef USE_UINT8
-        %template(DynamicByteArray) DynamicArray<uint8_t>;
-#endif
-#ifdef USE_INT16
-        %template(DynamicShortArray) DynamicArray<int16_t>;
-#endif
-#ifdef USE_UINT16
-        %template(DynamicWordArray) DynamicArray<uint16_t>;
-#endif
-#ifdef USE_INT32
-        %template(DynamicIntArray) DynamicArray<int32_t>;
-#endif
-#ifdef USE_UINT32
-        %template(DynamicUIntArray) DynamicArray<uint32_t>;
-#endif
-#ifdef USE_INT64
-        %template(DynamicLongArray) DynamicArray<int64_t>;
-#endif
-#ifdef USE_UINT64
-        %template(DynamicULongArray) DynamicArray<uint64_t>;
-#endif
-#ifdef USE_FLOAT32
-        %template(DynamicShortRealArray) DynamicArray<float32_t>;
-#endif
-#ifdef USE_FLOAT64
-        %template(DynamicRealArray) DynamicArray<float64_t>;
-#endif
-}
-/* Hash */
-%include <shogun/lib/Hash.h>
-
 %include <shogun/lib/Tokenizer.h>
 %include <shogun/lib/DelimiterTokenizer.h>
 %include <shogun/lib/NGramTokenizer.h>
-%include <shogun/lib/Cache.h>
 %include <shogun/lib/Signal.h>
 %include <shogun/lib/Time.h>
-%include <shogun/lib/Trie.h>
 %include <shogun/lib/Compressor.h>
 %include <shogun/lib/StructuredDataTypes.h>
 %include <shogun/lib/StructuredData.h>
 %include <shogun/lib/DynamicObjectArray.h>
+
+%define SGCONTAINER_NOVALUE_WRAPPER(TYPE)
+%feature("novaluewrapper") shogun::SGVector<TYPE>;
+%feature("novaluewrapper") shogun::SGMatrix<TYPE>;
+%feature("novaluewrapper") shogun::SGSparseVector<TYPE>;
+%feature("novaluewrapper") shogun::SGSparseMatrix<TYPE>;
+%template() shogun::SGVector<TYPE>;
+%template() shogun::SGMatrix<TYPE>;
+%template() shogun::SGSparseVector<TYPE>;
+%template() shogun::SGSparseMatrix<TYPE>;
+%enddef
+
+SGCONTAINER_NOVALUE_WRAPPER(bool)
+SGCONTAINER_NOVALUE_WRAPPER(char)
+SGCONTAINER_NOVALUE_WRAPPER(uint8_t)
+SGCONTAINER_NOVALUE_WRAPPER(uint16_t)
+SGCONTAINER_NOVALUE_WRAPPER(uint32_t)
+SGCONTAINER_NOVALUE_WRAPPER(uint64_t)
+SGCONTAINER_NOVALUE_WRAPPER(int16_t)
+SGCONTAINER_NOVALUE_WRAPPER(int32_t)
+SGCONTAINER_NOVALUE_WRAPPER(int64_t)
+SGCONTAINER_NOVALUE_WRAPPER(float32_t)
+SGCONTAINER_NOVALUE_WRAPPER(float64_t)
+SGCONTAINER_NOVALUE_WRAPPER(floatmax_t)
+SGCONTAINER_NOVALUE_WRAPPER(std::complex<double>)
```

Comment:
this removes sgcontainers from the interface but allows to still write the typemaps